src/expand.rs

```diff
@@ -371,7 +371,13 @@ fn transform_block(context: Context, sig: &mut Signature, block: &mut Block) {
                         let prefixed = Ident::new("__self", ident.span());
                         quote!(let #mutability #prefixed = #ident;)
                     } else {
-                        quote!(let #mutability #ident = #ident;)
+                        // If there is a `#[cfg(..)]` attribute that selectively
+                        // enables the parameter, forward it to the variable.
+                        let attrs = arg.attrs.iter();
+                        quote!(
+                            #(#attrs)*
+                            let #mutability #ident = #ident;
+                        )
                     }
                 } else {
                     let pat = &arg.pat;
```

Comment:
I'm not sure how to hit this branch -- I tried using a `DeconstructedStruct { value }` but it didn't parse.

Presumably the same attribute forwarding needs to be done here as well

Reply:
Try `#[cfg(any())] _: T` or `#[cfg(any())] (a, b): (T, T)`

Reply:
cool thanks, managed to simplify the test case